include/ros1_bridge/factory.hpp

```diff
@@ -69,7 +67,9 @@ class Factory : public FactoryInterface
     const std::string & topic_name,
     const rmw_qos_profile_t & qos_profile)
   {
-    return node->create_publisher<ROS2_T>(topic_name, qos_profile);
+    auto qos = rclcpp::QoS(rclcpp::KeepAll());
+    qos.get_rmw_qos_profile() = qos_profile;
```

Comment:
Why not replacing L70 with:

```cpp
rclcpp::QoS qos(rclcpp::QoSInitialization::from_rmw(qos_profile));
```

I think this method is used nowhere and it's no supposed to be used directly by the user, Maybe it could be just deleted. The method that is used is this other one:
https://github.com/ros2/ros1_bridge/blob/3015b901d0d6d3df4de810fa80757d180fa8297d/include/ros1_bridge/factory.hpp#L55-L62

Reply:
The reason it is like that is because `rclcpp::QoS qos(rclcpp::QoSInitialization::from_rmw(qos_profile));` only overrides the history and depth settings, it doesn't set any of the others, so I have to do `qos.get_rmw_qos_profile() = qos_profile;` no matter what, and in fact the `QoSInitialization`, in this case is completely unused.

Really, this function signature should probably be deprecated as well, since it takes the `rmw_qos_profile_t`, which would mean that this weird use of the `rclcpp::QoS` api could be removed next version. But that's part of a general need in these pr's to go and find secondary API's which use the `rmw_qos_profile_t` and deprecate those too, like in message filters or in services/actions, etc...

Maybe it could be removed? I was just updating every where I could find.